common/policies/orderer/implicit_orderer.go

```diff
@@ -0,0 +1,135 @@
+type IdentityDeserializerMock interface {
+	msp.IdentityDeserializer
+}
+
+//go:generate mockery -dir . -name IdentityMock -case underscore -output mocks/
+type IdentityMock interface {
+	msp.Identity
+}
+
+func NewPolicyProvider(
+	deserializer msp.IdentityDeserializer,
+	consensusType string,
+	consensusMetadata []byte,
+) policies.Provider {
+	return &policyProvider{
+		signaturePolicyProvider: &cauthdsl.ProviderFromStruct{Deserializer: deserializer},
+		consensusType:           consensusType,
+		consensusMetadata:       consensusMetadata,
+	}
+}
+
+func NewPolicyFromString(ruleName string) (*cb.ImplicitOrdererPolicy, error) {
+	logger.Debugf("Entry: ruleName=%s", ruleName)
+	ruleVal, exist := cb.ImplicitOrdererPolicy_Rule_value[ruleName]
+	if !exist {
+		return nil, errors.Errorf("ImplicitOrdererPolicy Rule '%v' not supported", ruleName)
+	}
+	return &cb.ImplicitOrdererPolicy{Rule: cb.ImplicitOrdererPolicy_Rule(ruleVal)}, nil
+}
+
+// NewPolicy creates a new policy based on the policy bytes
+func (p *policyProvider) NewPolicy(data []byte) (policies.Policy, proto.Message, error) {
+	definition := &cb.ImplicitOrdererPolicy{}
+	if err := proto.Unmarshal(data, definition); err != nil {
+		return nil, nil, errors.Wrap(err, "Error unmarshaling to ImplicitOrdererPolicy")
+	}
+	if definition.Rule != cb.ImplicitOrdererPolicy_SMARTBFT {
+		return nil, nil, errors.Errorf("ImplicitOrdererPolicy Rule '%v' not supported", definition.Rule)
+	}
+
+	smartbftMetadata := &protossmartbft.ConfigMetadata{}
+	if err := proto.Unmarshal(p.consensusMetadata, smartbftMetadata); err != nil {
+		return nil, nil, errors.Wrap(err, "failed to unmarshal smartbft metadata configuration")
+	}
+
+	var identities [][]byte
+	for _, consenter := range smartbftMetadata.Consenters {
+		id, err := crypto.SanitizeIdentity(consenter.Identity)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "failed to sanitize identity")
+		}
+		identities = append(identities, id)
+	}
+
+	q := computeQuorum(len(identities))
+
+	sigEnv := cauthdsl.SignedByNOutOfGivenIdentities(int32(q), identities)
+	sigPol, err := p.signaturePolicyProvider.NewPolicy(sigEnv)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to create SignedByNOutOfGivenRole signature policy")
+	}
+
+	ip := &implicitBFTPolicy{
+		sigPolicy:  sigPol,
+		quorumSize: q,
+	}
+
+	logger.Debugf("Successfully created an ImplicitOrdererPolicy with rule '%s', and %d out-of %d signature policy",
+		cb.ImplicitOrdererPolicy_Rule_name[int32(definition.Rule)], q, len(identities))
+
+	return ip, nil, nil
+}
+
+type implicitBFTPolicy struct {
+	sigPolicy  policies.Policy // N out of a set of explicit identities, taken from the consenter set
+	quorumSize int             // The BFT quorum size
+}
+
+func (ip *implicitBFTPolicy) Evaluate(signatureSet []*protoutil.SignedData) error {
+	logger.Debugf("Entry: signatureSet size: %d", len(signatureSet))
+
+	if len(signatureSet) < ip.quorumSize {
```

Comment:
so, why do we need this, if we have the N in the NoutOf Policy?

Reply:
for better diagnostics and debug.